File: tests/fixtures/typescript/sample-repo/src/duplicates.ts
// Hash-equality fixtures for the TS parser.
//
// Pairs marked `// duplicate-of: <name>` should produce identical BLAKE3 hashes.
// Pairs marked `// near-duplicate-of: <name>` should NOT, but should be close.

export function addA(x: number, y: number): number {
  return x + y;
}

export function addB(x: number, y: number): number {
  // duplicate-of: addA
  return x + y;
}

export function addC(x: number, y: number): number {
  // near-duplicate-of: addA
  return x - y;
}

export function addRenamed(a: number, b: number): number {
  // near-duplicate-of: addA
  // Identifier names changed; v1 normalization preserves names.
  return a + b;
}

export function addWithLiteral(x: number, y: number): number {
  // near-duplicate-of: addA
  return x + y + 0;
}

File: tests/fixtures/rust/sample-repo/src/duplicates.rs
// Hash-equality fixtures.
//
// `// duplicate-of: <name>` — should hash-equal under v1 normalization.
// `// near-duplicate-of: <name>` — should NOT hash-equal but be close.

pub fn add_a(x: i64, y: i64) -> i64 {
    x + y
}

pub fn add_b(x: i64, y: i64) -> i64 {
    // duplicate-of: add_a
    x + y
}

pub fn add_c(x: i64, y: i64) -> i64 {
    // near-duplicate-of: add_a
    x - y
}

pub fn add_renamed(a: i64, b: i64) -> i64 {
    // near-duplicate-of: add_a
    // Identifier names changed; v1 normalization preserves names.
    a + b
}

pub fn add_with_literal(x: i64, y: i64) -> i64 {
    // near-duplicate-of: add_a
    x + y + 0
}
